enhancements/proxy-passdown.md

```diff
@@ -0,0 +1,162 @@
+
+Working with a proxy is a feature that needs to be added to each
+operator separately, enabling the end-user story:
+"As an operator-admininstrator, I
+can deploy operators that respect a network proxy."
+
+For Cluster administrators to set up a network proxy:
+   1. Pods requiring network must respect proxy environment variables
+      `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY`
+   2. Operators must be provided environment variables
+   3. Operators must pass the environment variables to operands
+
+All of this is already possible but operator authors must implement this
+functionality themselves. This should be a "best practice"
+and documentation for this in our repository.
+
+## Proposal
+
+### Goals
+
+   1. Establish a best practice for operator authors:  Operators receive proxy
+      environment variables from the operator deployment, and pass proxy
+      environment variables to operands that require network.
+   2. Operator SDK features and documents to facilitate implementation of
+      proxy-friendly operators.
+
+### Non-Goals
+
+   1. It would be nice to encourage proxy friendly operators by
+      including that functionality as a badge or part of the maturity
+      model on operatorhub, but that is out of scope for this proposal.
+   2. Automatic scaffolding of proxy passdown cannot be done because the
+      logic must occur in operator-specific code.
+   3. It is beyond the scope of this proposal to set up trusted certs
+      required for https connections to a proxy.
+
+
+### How the Operator receives Env Vars
+
+In all cases, proxy environment variables to be passed to operands must
+be included on the Operator deployment. This can be handled by the
+administrator, or automatically set by OLM.
+
+### How the Operator passes Env vars to Operands
+
+For Go-based operators, set of functions will be created in operator-lib
+(and offered upstream to controller-runtime) to read the proxy variables
+from the Operator environment and inject these variables into each of
+the container specs of various kubernetes objects. Operator authors will
+be responsible for invoking these functions. We will need functions for:
+
+   0. operatorlib.ReadProxyVarsFromEnv() // Helper function
+   1. operatorlib.AddProxyToDeployment(dep *appsv1.Deployment)
+   2. operatorlib.AddProxyToDaemonSets(daem *appsv1.DaemonSet)
+   3. operatorlib.AddProxyToReplicaSets(rep *appsv1.ReplicaSet)
+   4. operatorlib.AddProxyToPods(pod *appsv1.Pod)
+
+Simplified Code:
+
+```go
+// Read environment variables for "HTTP_PROXY, HTTPS_PROXY, and
+// NO_PROXY"
+func ReadProxyVarsFromEnv() ([]corev1.EnvVar) {
+   httpProxyEnvVar := "HTTP_PROXY"
+   httpproxy := os.Getenv("HTTP_PROXY")
+   return []corev1.EnvVar{
+       {
+           Name:  httpProxyEnvVar,
+           Value: httpproxy,
+       },
+       {
+           Name:  strings.ToLower(httpProxyEnvVar),
+           Value: httpproxy,
+       },
+    }
+}
+
+func AddProxyToDeployment(dep *appsv1.Deployment) {
```

Comment:
There has been some discussion that perhaps adding proxy envs to appsv1-based objects is moving in the wrong direction, since this won't work with the future of Unstructured objects and server side apply. I'll talk about this with others and report back here. `ReadProxyVarsFromEnv` would still be useful.

Reply:
After discussing this, we have decided not to include the specific object helpers at this time. Instead, we will have docs and examples using the `ReadProxyVarsFromEnv` helper.